tests/unit/ast/test_ast_dict.py

```diff
@@ -142,21 +142,1039 @@ def test() -> int128:
 
 # strip source annotations like lineno, we don't care for inspecting
 # the analysis result
-def _strip_source_annotations(dict_node):
-    to_strip = NODE_SRC_ATTRIBUTES + ("node_id",)
+def _strip_source_annotations(dict_node, to_strip):
     if isinstance(dict_node, dict):
         for k in list(dict_node.keys()):
             if k in to_strip:
                 del dict_node[k]
                 continue
-            _strip_source_annotations(dict_node[k])
+            if "decl_node" not in k:
+                _strip_source_annotations(dict_node[k], to_strip)
     elif isinstance(dict_node, list):
         for child in dict_node:
-            _strip_source_annotations(child)
+            _strip_source_annotations(child, to_strip)
+
+
+def test_output_type_info(make_input_bundle, chdir_tmp_path):
+    # test type info is output in the ast dict
+    # test different, complex types, and test import info is also output
+    lib1 = """
+struct Foo:
+    x: uint256
+
+event Bar:
+    pass
+
+struct Baz:
+    x: decimal
+    y: Bytes[20]
+    z: String[32]
+    w: uint256
+    u: address
+
+interface Qux:
+    def return_tuple() -> (Foo[1], uint256): nonpayable
+
+foo_var: Foo
+sarray_var: Foo[1]
+darray_var: DynArray[Foo, 5]
+interface_var: Qux
+
+hashmap_var: HashMap[address, Foo]
+
+sarray_var2: uint256[2]
+darray_var2: DynArray[uint256, 5]
+
+@internal
+def foo():
+    t: uint256 = max_value(uint256)
+    u: int24 = empty(int24)
+
+    self.foo_var = empty(Foo)
+    self.sarray_var[0] = empty(Foo)
+    self.darray_var[1] = empty(Foo)
+
+    self.sarray_var, t = self.interface_var.return_tuple()
+
+@external
+def bar():
+    s: bytes24 = empty(bytes24)
+    """
+
+    main = """
+import lib1
+
+initializes: lib1
+
+@internal
+def foo():
+    lib1.foo()
+    log lib1.Bar()
+    s: lib1.Foo = empty(lib1.Foo)
+    """
+
+    input_bundle = make_input_bundle({"lib1.vy": lib1, "main.vy": main})
+
+    lib1_file = input_bundle.load_file("lib1.vy")
+    out = compiler.compile_from_file_input(
+        lib1_file, input_bundle=input_bundle, output_formats=["annotated_ast_dict"]
+    )
+    lib1_ast = out["annotated_ast_dict"]["ast"]
+    lib1_sha256sum = lib1_ast.pop("source_sha256sum")
+    assert lib1_sha256sum == lib1_file.sha256sum
+    to_strip = NODE_SRC_ATTRIBUTES + ("resolved_path", "variable_reads", "variable_writes")
+    _strip_source_annotations(lib1_ast, to_strip=to_strip)
+
+    main_file = input_bundle.load_file("main.vy")
+    out = compiler.compile_from_file_input(
+        main_file, input_bundle=input_bundle, output_formats=["annotated_ast_dict"]
+    )
+    main_ast = out["annotated_ast_dict"]["ast"]
+    main_sha256sum = main_ast.pop("source_sha256sum")
+    assert main_sha256sum == main_file.sha256sum
+    _strip_source_annotations(main_ast, to_strip=to_strip)
+
+    assert main_ast == {
```

Comment:
i would like to stay away from these types of tests where we copy huge chunks of the compiler's output and hope we didn't miss anything..

we should aim to create multiple small, easily verifiable tests

Reply:
same, but it was the most straightforward way to test for now. i'll add a note to refactor